### PR TITLE
vcsim: untie datastore capacity from local fs

### LINKED ISSUE #2390: vcsim should provide a way to set the initial capacity for a datastore
When running `vcsim`, I've noticed that when you're looking at cluster-level usage, the amount of free space can vary depending on where you're running `vcsim`! Doing some digging, it looks like `vcsim` makes a syscall to the host system to get the amount of available disk space and uses that as the amount of unused capacity for the underlying datastore. Normally this isn't an issue, but if you happen to run `vcsim` on a machine without a lot of free disk space then it limits what you can do with the simulator.

One way around the issue is to add more datastores, but that isn't really representative of the environment we're trying to simulate. It would be better if there was a way to set the amount of initial capacity in the datastore as either a parameter on the `vcsim` command or as an exposed parameter in `simulator.Model`.

### PR DESCRIPTION
**Pull request by dougm**
Using the current local filesystem to set datastore capacity is not useful and vcsim uses very little actual disk space.
The simulator.VirtualMachine impl already adjusts datastore FreeSpace when VirtualDisks, Snapshots, etc, are added and removed.
This change removes the os Statfs and sets capacity to 10TB by default.

The default disk size for VMs created by the Model has been increased from 1K to 10G

Fixes #2390

### CODE REVIEW

**Comment by embano1** [comment]
Hey @dougm sorry for my delayed review. Will do thorough review early next week, but good work!

QQ: looks like the usage of this env var `VCSIM_DATASTORE_CAPACITY` is not documented (only in this PR). How about adding this to `vcsim` docs?

File: simulator/datastore.go [hunk 1]
```diff
@@ -65,19 +66,15 @@ func parseDatastorePath(dsPath string) (*object.DatastorePath, types.BaseMethodF
...
 	}
 
 	info := ds.Info.GetDatastoreInfo()
 
-	now := time.Now()
-
-	info.Timestamp = &now
-	info.MaxMemoryFileSize = info.FreeSpace
-	info.MaxFileSize = info.FreeSpace
+	info.Timestamp = types.NewTime(time.Now())
```
Comment (embano1) [inline simulator/datastore.go:77]: For testing (perhaps in future) and deterministic behavior, should we make time mockable, e.g. via global timeFn or a clock (https://github.com/benbjohnson/clock)?

File: simulator/host_datastore_system.go [hunk 2]
```diff
@@ -83,11 +84,28 @@ func (dss *HostDatastoreSystem) add(ctx *Context, ds *Datastore) *soap.Fault {
...
-	browser.Datastore = dss.Datastore
-	ds.Browser = Map.Put(browser).Reference()
+	if Map.Get(ds.Self) == nil {
+		browser := &HostDatastoreBrowser{}
+		browser.Datastore = dss.Datastore
+		ds.Browser = Map.Put(browser).Reference()
 
-	folderPutChild(ctx, folder, ds)
+		var capacity units.ByteSize
+		b := os.Getenv("VCSIM_DATASTORE_CAPACITY")
```
Comment (embano1) [inline simulator/host_datastore_system.go:93]: Can you please document `"VCSIM_DATASTORE_CAPACITY"` in the README?

File: simulator/host_datastore_system.go [hunk 3]
```diff
@@ -83,11 +84,28 @@ func (dss *HostDatastoreSystem) add(ctx *Context, ds *Datastore) *soap.Fault {
...
+		if b == "" {
+			b = "1TB"
+		}
+		capacity.Set(b)
```
Comment (embano1) [inline simulator/host_datastore_system.go:97]: Ignoring error is fine here?

File: simulator/host_datastore_system.go [hunk 4]
```diff
@@ -83,11 +84,28 @@ func (dss *HostDatastoreSystem) add(ctx *Context, ds *Datastore) *soap.Fault {
...
+
+		ds.Summary.Capacity = int64(capacity)
+		ds.Summary.FreeSpace = ds.Summary.Capacity
+
+		info := ds.Info.GetDatastoreInfo()
```
Comment (embano1) [inline simulator/host_datastore_system.go:102]: `info` is shadowed here, intentionally?

File: simulator/datastore.go (hunk 1 above)
Comment (dougm) [reply simulator/datastore.go:77]: Not sure we need testing of the time value in this case, other than just `Timestamp != nil`

File: simulator/host_datastore_system.go (hunk 2 above)
Comment (dougm) [reply simulator/host_datastore_system.go:93]: Not sure we should use an env var here at all.  Tempted to leave it hardcoded as `1TB` (give or take) and find a better way to configure the default if we find we need to.  Any thoughts @protochron ?

File: simulator/host_datastore_system.go (hunk 3 above)
Comment (dougm) [reply simulator/host_datastore_system.go:97]: Ah yeah, we should just `panic()` on error here I think.

File: simulator/host_datastore_system.go (hunk 4 above)
Comment (dougm) [reply simulator/host_datastore_system.go:102]: Whoops, no, I'll remove this line.

File: simulator/datastore.go (hunk 1 above)
Comment (embano1) [reply simulator/datastore.go:77]: Fine with me.

File: simulator/host_datastore_system.go (hunk 2 above)
Comment (protochron) [reply simulator/host_datastore_system.go:93]: I'm fine with a hardcoded value, though maybe use something like `10TB` and call it a day?

**Comment by protochron** [comment]
Sorry for the delayed response on my end, I was on vacation for awhile.

I think the approach here looks sound. Being able to modify the default datastore size via an environment variable would be useful, but possibly not needed as long as the default size isn't too small.

**Comment by dougm** [comment]
@protochron hope you had a good vacation!  Sounds good to me, 10TB it is.  We can address the use case(s) of being able to change that if/when it comes up.   cc @embano1

**Review by embano1 — APPROVED** [review]

**Comment by embano1** [comment]
Thx @dougm, please just update the PR description to reflect the last changes.

**Comment by dougm** [comment]
@embano1 good catch, updated commit message too

**Review by embano1 — APPROVED** [review]
LGTM
